backend/src/platform/api/routes.py

```
@@ -70,20 +67,21 @@
 
 
-def _principal_from_request(request: Request) -> Principal:
-    principal = getattr(request.state, "principal", None)
-    if not principal:
-        raise PermissionError("missing principal context")
-    return principal
+def _principal_id_from_request(request: Request) -> str:
+    """Extract principal_id from request state."""
+    principal_id = getattr(request.state, "principal_id", None)
+    if not principal_id:
+        raise PermissionError("missing principal_id context")
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)